### DISCUSSION TITLE
Configuring Ext Clock

### DISCUSSION THREAD

**Discussion by nilsbuc** [PandABlocks/PandABlocks.github.io#9] [Q&A] [answered]
Hey,
can someone help me configure an external clock signal to the PandAbox. Right now I tried to hook the ext clock input to a signal generator. But before even trying different functions, the PandAbox already gives me the message "clock select error", when I choose "sma clock" as "clock source".

Thank you
nils

**Comment by Araneidae** [comment]
@thomascobb, can we please enable "Discussions" so that these queries can be handled there.

**Comment by glennchid** [comment]
Hi Nils,
What is the version of the firmware you are running, and what the frequency of the external clock you are trying to use?

**Reply by nilsbuc** [reply]
Hey glennchid,
I'm using the panda-fpga@PandABox-fmc_acq430-3.0.zpg firmware. At the moment I don't have a specific frequency I'm trying to use. I want the clock-blocks to have the same phase as the signals I put in to the PandAbox, so that I can capture Data at the right moment.

**Reply by glennchid** [reply, marked as answer]
Is the PLL showing as locked in the System block? I think a frequency of 125 MHz is assumed for the PLLs. If you are using something close to 125 MHz, I can try it on one of our PandABoxes. If the PLL does not lock it will refuse to switch to that clock, and give the 'clock select error' in the Health field.

**Reply by nilsbuc** [reply]
That was it, I was using a frequency that was too low, with 125 MHz the PPL goes to 1. Do you also know what the voltage range is at the clock input, cause I couldn't find anything in the user manual. 
Thank you.

**Reply by Araneidae** [reply]
> Do you also know what the voltage range is at the clock input, cause I couldn't find anything in the user manual.

This is a good question and might even be an issue to raise against the documentation.  I know that 0dBm (0.6 Vpp or 0.2 Vrms) is a common safe and adequate value, but the appropriate range will depend on the input driver on the PandA board.  Probably @glennchid can comment.

**Reply by glennchid** [reply]
I don't know. The comparator is an [LMH7220](https://www.ti.com/lit/ds/symlink/lmh7220.pdf).

It claims to have an absolute maximum input voltage of +/- 13V. In the lab we are using a clock generator set to 1.2 V.

**Reply by Araneidae** [reply]
This is one of those cases where the input circuitry is also relevant.  In particular, the LMH7220 is powered with a 3.3V rail, and probably more to the point is protected by diodes (and there is a 100Ω termination).  I don't have the part numbers of the protection diodes to hand, but if the signal is high enough to drive the diodes the clock will become distorted and more noisy.

So the trade-offs are:
* Too low a signal level will produce a noisy clock
* Too high a signal will result in a distorted and noisy clock
* Stupidly high signal levels will eventually break something, but it looks as if it's reasonably well protected.

Unfortunately I don't know what an appropriate signal level is.  1.2Vpp sounds fine to me; at 100Ω that's about +2.5 dBm.
